pkg/test/utils.go

```diff
@@ -65,15 +66,35 @@ func MustReadFile(t *testing.T, p string) []byte {
 	return b
 }
 
+func extractEventsWithURLPrefix(allEvents, urlPrefix string) string {
+	eventStrings := make([]string, 0)
+	events := strings.Split(allEvents, "---")
+	for _, event := range events {
+		processed := strings.TrimSpace(event)
+		for _, m := range []string{"GET", "POST", "PUT", "PATCH", "DELETE"} {
+			processed = strings.TrimPrefix(processed, m)
+			processed = strings.TrimSpace(processed)
+		}
+		if strings.HasPrefix(processed, urlPrefix) {
+			if len(eventStrings) == 0 {
+				event = strings.TrimPrefix(event, "\n\n")
+			}
+			eventStrings = append(eventStrings, event)
+		}
+	}
+	return strings.Join(eventStrings, "---")
+}
+
 // CompareGoldenFile performs a file comparison for a golden test.
-func CompareGoldenFile(t *testing.T, p string, got string, normalizers ...func(s string) string) {
+func CompareGoldenFile(t *testing.T, path, got string, normalizers ...func(s string) string) {
 	writeGoldenOutput := os.Getenv("WRITE_GOLDEN_OUTPUT") != ""
 
 	for _, normalizer := range normalizers {
 		got = normalizer(got)
 	}
+	gotForComp := got
 
-	wantBytes, err := os.ReadFile(p)
+	wantBytes, err := os.ReadFile(path)
```

Comment:
Amusing counterexample to the "no abbreviations" guideline :-)  We tend to use `p` instead of `path` so we don't mask the `path` package.  But normally we should be using `filepath` instead, so maybe I should follow the "no abbreviations" rule for `path` also :-)

Reply:
Good point! Will change it back.